specs/TitleBar/titlebar-functional-spec.md

```diff
@@ -0,0 +1,479 @@
+
+
+## TitleBar.IconSource property
+
+Gets or sets the TitleBar's icon.
+
+```cs
+public IconSource IconSource { get; set; }
+```
+
+### Property Value
+Represents the icon associated with the `TitleBar`. The default is `null`.
+
+
+## TitleBar.Content property
+
+Gets and sets elements within the TitleBar's Content column.
+
+```cs
+public UIElement Content { get; set; }
+```
+
+### Property Value
+Represents the Content property of the `TitleBar` control which can be populated in both XAML markup and code. 
+
+This is typically used to populate controls such as `AutoSuggestBox`. The default is `null`.
+
+### Remarks
+The content property of `Content` can only be set once. Elements are center aligned by default.
+
+
+## TitleBar.Footer property
+
+Gets and sets elements within the TitleBar's Footer column.
+
+```cs
+public UIElement Footer { get; set; }
+```
+
+### Property Value
+Represents the Footer property of the `TitleBar` control which can be populated in both XAML markup and code. 
+
+This is typically used to populate controls such as `PersonPicture` and the "More" `AppBarButton`. The default is `null`.
+
+### Remarks
+The content property of `Content` can only be set once. Elements are right aligned by default.
+
+
+## TitleBar.IsBackButtonVisible property
+
+Gets and sets TitleBar's BackButton visiblity.
+
+```cs
+public Boolean IsBackButtonVisible { get; set; }
+```
+
+### Property Value
+Represents the visiblity of TitleBar's BackButton. The default is `false`.
+
+
+## TitleBar.IsBackButtonEnabled property
+
+Gets and sets BackButton's IsEnabled property.
+
+```cs
+public Boolean IsBackButtonEnabled { get; set; }
+```
+
+### Property Value
+Represents the IsEnabled property of TitleBar's BackButton. The default is `true`.
+
+### Remarks
+This property is typically bound to an accompanying NavFrame's `CanGoBack` property:
+`IsBackButtonEnabled="{x:Bind NavFrame.CanGoBack}"`
+
+## TitleBar.IsPaneToggleButtonVisible property
+
+Gets and sets TitleBar's PaneToggleButton visiblity.
+
+```cs
+public Boolean IsPaneToggleButtonVisible { get; set; }
+```
+
+### Property Value
+Represents the visiblity of TitleBar's PaneToggleButton. The default is `false`.
+
+### Remarks
+TitleBar's PaneToggleButton is only used with NavigationView and when NavigationView is in `Minimal` DisplayMode. 
+In other display mode scenarios, the PaneToggleButton is owned and handled by NavigationView. 
+
+
+## TitleBar.BackRequested event
+
+Occurs whenever BackButton is clicked.
+
+C#
+```cs
+public event TypedEventHandler<TitleBar, Object> BackRequested;
+```
+
+### Remark
+This event is raised when internal BackButton raises a Click event.
+
+
+## TitleBar.PaneToggleRequested event
+
+Occurs whenever PaneToggleButton is clicked.
+
+C#
+```cs
+public event TypedEventHandler<TitleBar, Object> PaneToggleRequested;
+```
+
+### Remark
+This event is raised when internal PaneToggleButton raises a Click event.
+
+## TitleBarTemplateSettings Class
+<!-- TODO -->
```

Comment:
I'd like to see information on this class. I had pointed out in https://github.com/microsoft/microsoft-ui-xaml/issues/9700 that this was an odd way to set the icon here, rather than just setting the icon directly on the window or titlebar instance. It feels a bit like this is just internal plumbing that is bubbling up into the public API.

Reply:
The TitleBarTemplateSettings class would not be used to set the Icon Property. It was added originally to implement interactive / non-interactive regions in the TitleBar. I will most likely be removing it, as it is no longer needed.

Reply:
Oops, I misspoke in my previous comment. The Icon in TitleBar will continue to be applied through TitleBarTemplateSettings.IconElement. By best practice, we should not be modifying xaml elements in codebehind, should the element be removed in re-templating.

Reply:
This confuses me. Normally templatesettings classes are used to simplify binding things into the template, and not as the main way to set a property (which would also make it harder to set from xaml). Reference: https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/template-settings-classes#the-scenario-for-templatesettings-classes